### PR TITLE
feat: Completely separate ingestion for replay events

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

We made good progress and have now got separately built events for the new blob ingestion flow. We just need to hook it all up now

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: .run/PostHog.run.xml [hunk 1]
```diff
@@ -32,6 +34,11 @@
         <ENTRY IS_ENABLED="true" PARSER="runconfig" IS_EXECUTABLE="false" />
       </ENTRIES>
     </EXTENSION>
+    <EXTENSION ID="software.aws.toolkits.jetbrains.core.execution.PythonAwsConnectionExtension">
+      <option name="credential" />
+      <option name="region" />
+      <option name="useCurrentConnection" value="false" />
+    </EXTENSION>
```
Comment (pauldambra) [inline .run/PostHog.run.xml:41]:
```suggestion
```

**Review by pauldambra — COMMENTED** [review]
finishing the review so I can apply a change 🙃

File: posthog/api/capture.py [hunk 2]
```diff
@@ -155,7 +163,14 @@ def log_event(data: Dict, event_name: str, partition_key: Optional[str]):
 
     # TODO: Handle Kafka being unavailable with exponential backoff retries
     try:
-        future = KafkaProducer().produce(topic=kafka_topic, data=data, key=partition_key)
+        if event_name in SESSION_RECORDING_DEDICATED_KAFKA_EVENTS:
+            producer = sessionRecordingKafkaProducer()
+
+            if not producer:
+                raise Exception("Session recording kafka producer not available")
```
[the remark this answers is not in the document]
Comment (pauldambra) [reply posthog/api/capture.py:170]: Ah, no, we're just saying it's invalid to set the one setting without setting the other

[the remark this answers is not in the document]
Comment (benjackwhite) [reply posthog/api/capture.py:170]: Non-throwing would be harder I think and I'd rather throw as this should _never_ happen

**Comment by pauldambra** [comment]
writes to minio locally but doesn't playback... no errors though

**Review by pauldambra — APPROVED** [review]
LGTM

**Pull request merged**